compiler/passes/normalize.cpp

```diff
@@ -1873,14 +1883,15 @@ static void normVarTypeInference(DefExpr* defExpr) {
         argExpr->baseExpr->replace(new UnresolvedSymExpr("init"));
 
         // Add _mt and _this (insert at head in reverse order)
-        argExpr->insertAtHead(var);
+        argExpr->insertAtHead(new NamedExpr("this", new SymExpr(var)));
```

Comment:
It troubles me that we are going back to the use of NamedExpr to signal things.
I don't understand this issue very well but it's hard for to understand why it's necessary.